### PR TITLE
chore: Migrations for AuthModule, AsmModule, KymaModule and CDC.

### LINKED ISSUE #9622: Auth/Session management deprecations
First, see the PR message: https://github.com/SAP/spartacus/pull/9297

- [x] `integration-libs/cdc/src/auth/facade/cdc-auth.service.ts` - changed constructor
- [x] `integration-libs/cdc/src/auth/services/user-authentication/cdc-user-authentication-token.service.ts` - not part of public API
- [x] `integration-libs/cdc/src/auth/store/effects/cdc-user-token.effect.ts` - changed constructor (EDIT @Platonn : not part of public API)
- [x] `projects/core/src/auth/index.ts` - removed apis
- [x] `projects/core/src/auth/services/index.ts` - removed
- [x] `projects/core/src/auth/store/**` - removed
- [x] `projects/core/src/auth/client-auth/services/client-authentication-token.service.ts` - changed constructor (EDIT @Platonn : before 3.0 it was not part of public API)
- [x] `projects/core/src/auth/user-auth/services/auth-redirect.service.ts` - changed constructor
- [x] `projects/core/src/auth/user-auth/guards/index.ts` - removed api
- [x] `projects/core/src/auth/http-interceptors/index.ts` - removed
- [x] `projects/core/src/kyma/**` - removed
- [x] `KymaService`, `KymaConfig`, `defaultKymaConfig` - removed / renamed
- [x] `projects/core/src/asm/index.ts` - removed apis
- [x] `projects/core/src/asm/services/index.ts` - removed apis
- [x] `projects/core/src/auth/config/auth-config.ts`, `projects/core/src/auth/config/default-auth-config.ts` - removed files
- [x] `projects/core/src/auth/user-auth/config/auth-config.ts`, `projects/core/src/auth/user-auth/config/default-auth-config.ts` - added files
- [x] `projects/core/src/auth/store/index.ts` renamed to `projects/core/src/auth/client-auth/store/index.ts`

### PR DESCRIPTION
**Pull request by marlass**
Closes #9622

### CODE REVIEW

File: docs/migration/3_0.md [hunk 1]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+
+#### Config
+
+- `AuthConfig` no longer extends `OccConfig`.
+- `login` and `revoke` endpoints were removed from `OccConfig`. `login` endpoint is now available under `tokenEndpoint` property in `AuthConfig`. `revoke` endpoint is available under `revokeEndpoint` property in `AuthConfig`.
+- `storageSync` configuration for `auth` branch in ngrx store was removed. State of token, userId is now synchronized with `AuthStatePersistenceService`. Override this service if you want to sync more properties to localStorage (eg. `refresh_token`).
+
+### Kyma module
+
+Kyma module were removed with all it's code. We expose the same functionality through configuration of auth.
```
Comment (Platonn) [inline docs/migration/3_0.md]: For easier search-ability:
```suggestion
### KymaModule

`KymaModule` were removed with all it's code. We expose the same functionality through configuration of auth.
```

File: docs/migration/3_0.md [hunk 2]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+- `UserTokenState` was removed. Data related to user token are no longer stored in ngrx store. User token is stored in `AuthStorageService` and user id is stored in `UserIdService`.
+- `AUTH_FEATURE` was removed. Client token state is under the key from `CLIENT_AUTH_FEATURE` variable.
+- `CLIENT_TOKEN_DATA` variable value changed to `[Client auth] Client Token Data`. Previously it was `[Auth] Client Token Data`.
+- `AuthActions` now only contains `Login` and `Logout` action class and `LOGIN`, `LOGOUT` variables with action type. Actions related to client token (`LoadClientToken...`) are now available under `ClientAuthActions` export. Constant `LOGOUT_CUSTOMER_SUPPORT_AGENT` is available in `AsmActions` export. Actions related to user token were removed. Instead initialize user token load, refresh or revoke with methods exposed in `AuthService` and `OAuthLibWrapperService`.
+
+#### Models
+
+- `UserToken` interface was replaced with `AuthToken` interface. New interface contains different properties than the previous to match requirements of `angular-oauth2-oidc` library.
```
Comment (Platonn) [inline docs/migration/3_0.md:121]:
```suggestion
- `UserToken` interface was replaced with `AuthToken` interface. New interface contains different properties than the previous to match requirements of `angular-oauth2-oidc` library.
- `AuthenticationToken` interface was removed. Use `AuthToken` instead.
```

File: docs/migration/3_0.md [hunk 3]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+
```
Comment (Platonn) [inline docs/migration/3_0.md:147]: Let's mention another model that was removed `export type AuthenticationToken = UserToken | ClientToken;`
```suggestion
- `AuthenticationToken` interface was removed. Replacement: ...
```

File: docs/migration/3_0.md [hunk 4]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+- Effects in `AsmModule` now instead of `makeErrorSerializable` use `normalizeHttpError` for error transformation.
+- `storageSync` configuration for `asm` branch in ngrx store was removed. State of ASM UI, tokens is now synchronized with `AsmStatePersistenceService`.
+- `CSAGENT_TOKEN_DATA` variable was removed.
+- `AsmState.csagentToken` was removed. Token is now stored in `AuthStorageService`. Check `AsmAuthStorageService.getTokenTarget` to validate if the token belongs to the CS agent.
+- `AsmActions` no longer contains actions related to customer agent token. Instead interact directly with `CsAgentAuthService`.
+- `CustomerSupportAgentTokenInterceptor` interceptor was removed. Token and error handling for CS agent requests are now handled by `AuthInterceptor` and `AsmAuthHttpHeaderService`.
+
+#### AsmAuthService
+
+Service was renamed to `CsAgentAuthService`. `AsmAuthService` is now responsible for making `AuthService` aware of ASM and adjusts it for CS agent support.
```
Comment (Platonn) [inline docs/migration/3_0.md:204]: What about adding removed-api-migration for `AsmAuthService` with custom comment:
The `AsmAuthService` was renamed to `CsAgentAuthService`. That said, the new service `AsmAuthService` was introduced in 3.0, which has a different purpose. For more see 3.0 migration docs.

File: docs/migration/3_0.md [hunk 5]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+Then you can access OpenId token with `OAuthLibWrapperService.getIdToken` method. For more options related to OpenId token look into `angular-oauth2-oidc` library documentation.
+
+### ASM module refactor
+
+- `getCustomerSupportAgentTokenState`, `getCustomerSupportAgentToken` and `getCustomerSupportAgentTokenLoading` were removed from `AsmSelectors`. To get token use `AuthStorageService.getToken` and `AsmAuthStorageService.getTokenTarget` to check if it belongs to CS agent.
+- Effects in `AsmModule` now instead of `makeErrorSerializable` use `normalizeHttpError` for error transformation.
+- `storageSync` configuration for `asm` branch in ngrx store was removed. State of ASM UI, tokens is now synchronized with `AsmStatePersistenceService`.
+- `CSAGENT_TOKEN_DATA` variable was removed.
+- `AsmState.csagentToken` was removed. Token is now stored in `AuthStorageService`. Check `AsmAuthStorageService.getTokenTarget` to validate if the token belongs to the CS agent.
+- `AsmActions` no longer contains actions related to customer agent token. Instead interact directly with `CsAgentAuthService`.
```
Comment (Platonn) [inline docs/migration/3_0.md]: Do you think it's worth listing them here for searchability? I believe docs are 1st source of truth and automated migrations are only follow up.
LoadCustomerSupportAgentToken
LoadCustomerSupportAgentTokenFail
LoadCustomerSupportAgentTokenSuccess

File: docs/migration/3_0.md [hunk 6]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+- `AuthState` was removed. State related to client token was moved to `ClientAuthState`. Data related to user token are stored in `AuthStorageService` and `UserIdService`.
+- `UserTokenState` was removed. Data related to user token are no longer stored in ngrx store. User token is stored in `AuthStorageService` and user id is stored in `UserIdService`.
+- `AUTH_FEATURE` was removed. Client token state is under the key from `CLIENT_AUTH_FEATURE` variable.
+- `CLIENT_TOKEN_DATA` variable value changed to `[Client auth] Client Token Data`. Previously it was `[Auth] Client Token Data`.
+- `AuthActions` now only contains `Login` and `Logout` action class and `LOGIN`, `LOGOUT` variables with action type. Actions related to client token (`LoadClientToken...`) are now available under `ClientAuthActions` export. Constant `LOGOUT_CUSTOMER_SUPPORT_AGENT` is available in `AsmActions` export. Actions related to user token were removed. Instead initialize user token load, refresh or revoke with methods exposed in `AuthService` and `OAuthLibWrapperService`.
```
Comment (Platonn) [inline docs/migration/3_0.md]: `LoadClientToken...` 

Should we list them for better searchabilty?

File: docs/migration/3_0.md [hunk 7]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+- `AuthService.authorize` was renamed to `loginWithCredentials`. It returns Promise that resolves after login procedure completes. Instead of dispatching action the method now invokes login method from the OAuth library and sets correct userId, dispatches `Login` action and redirects to previously visited page.
+- `AuthService.getOccUserId` was removed from `AuthService`. Use `UserIdService.getUserId` method instead. It is the direct replacement.
+- `AuthService.invokeWithUserId` was moved to `UserIdService`. It is available under the same name.
+- `AuthService.getUserToken` was removed. To check if user is logged in use `isUserLoggedIn` and to get user id use `UserIdService.getUserId`. If you need access to tokens then use `AuthStorageService.getToken`.
+- `AuthService.refreshUserToken` was moved and renamed to `OAuthLibWrapperService.refreshToken`. The behavior changed as well. It not only dispatches action, but performs complete refresh token procedure from OAuth library.
+- `AuthService.authorizeWithToken` was removed. Instead you can create object of the shape 'AuthToken' and pass to 'AuthStorageService.setToken'.
```
Comment (Platonn) [inline docs/migration/3_0.md]: Nit: formatting?
```suggestion
- `AuthService.authorizeWithToken` was removed. Instead you can create object of the shape `AuthToken` and pass to `AuthStorageService.setToken`.
```

File: docs/migration/3_0.md [hunk 8]
```diff
@@ -102,6 +102,99 @@ There can be more than one cart entry with the same product code. So now they ar
...
+
+#### AsmAuthService
+
+Service was renamed to `CsAgentAuthService`. `AsmAuthService` is now responsible for making `AuthService` aware of ASM and adjusts it for CS agent support.
+
+- `AsmAuthService.authorizeCustomerSupportAgent` was moved to `CsAgentAuthService`. It now performs full login flow for CS agent and resolves when it completes.
+- `AsmAuthService.startCustomerEmulationSession` was moved to `CsAgentAuthService`. Behavior haven't changed.
+- `AsmAuthService.isCustomerEmulationToken` was removed. To check token use `AuthStorageService.getToken` and to check if it belongs to CS agent use `AsmAuthStorageService.getTokenTarget`.
+- `AsmAuthService.getCustomerSupportAgentToken` was removed. To check token use `AuthStorageService.getToken` and to check if it belongs to CS agent use `AsmAuthStorageService.getTokenTarget`.
+- `AsmAuthService.getCustomerSupportAgentTokenLoading` was moved to `CsAgentAuthService`. Warning! It is not implemented there yet.
```
Comment (Platonn) [inline docs/migration/3_0.md:210]: 👀

Comment (marlass) [reply docs/migration/3_0.md:210]: There is a ticket for that and we might able to complete that before stable release ;)

**Review by Platonn — APPROVED** [review]

**Pull request merged**